Dockerfile

```
@@ -1,7 +1,10 @@
-FROM ocaml/opam:alpine-3.21-ocaml-5.2 AS build
+FROM alpine:3.21 AS build
 
 # Install system dependencies
-RUN sudo apk -U upgrade --no-cache && sudo apk add --no-cache \
+RUN apk -U upgrade --no-cache && apk add --no-cache \
```

```suggestion
RUN apk -U upgrade && apk add \
```

Here you can keep the cache as this is a multi-staged build. This will avoid downloading the package index twice.